CMakeLists.txt

```diff
@@ -28,27 +26,64 @@ if(HIGHFIVE_PARALLEL_HDF5)
 set(HDF5_PREFER_PARALLEL TRUE)
 endif()
 
-find_package(HDF5 REQUIRED)
+find_package(HDF5 QUIET REQUIRED)
 
 set(Boost_NO_BOOST_CMAKE TRUE)
 
 
 if(USE_BOOST)
-    find_package(Boost 1.41 REQUIRED COMPONENTS system unit_test_framework)
-    add_definitions(-DH5_USE_BOOST )
+  find_package(Boost 1.41 QUIET REQUIRED COMPONENTS system unit_test_framework)
+  add_definitions(-DH5_USE_BOOST )
 endif()
 
 if(HIGHFIVE_PARALLEL_HDF5)
-    find_package(MPI)
+  find_package(MPI)
 endif()
 
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/highfive
 		DESTINATION ${INCLUDE_INSTALL_DIR})
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include ${HDF5_INCLUDE_DIRS} ${Boost_INCLUDE_DIR} ${MPI_C_INCLUDE_PATH})
+# cmake target install
+if(CMAKE_MAJOR_VERSION GREATER 2) # No export(EXPORT) in CMake 2
```

Comment:
I've done the split.

I don't think the variables exported and from the cmakeConfig files are any more intrusive than a FindFoo.cmake borrowed from somewhere, with two added advantages:
- The Config files are always in sync with the version of the software that produced them.
- Since they use target properties to configure include and link requirements instead of variables, your client code is simpler and your namespace is less polluted than with FindFoo.cmake files.
Regarding modified variables, I don't get what you mean by that, I don't see parent scope variables being modified in a Config.cmake file.

In any case, the config,cmake file generation is not the most important part of this PR. It's the target_include_directories what's really useful for client code because then you get all the include dependencies automatically (including the transitive ones) by simple adding HighFive to your target_link_libraries.